src/scripts/api.js

```
@@ -140,255 +1,341 @@
-            ]
+        } else {
+            console.info(`CREDS INFO: Data incomplete! Retrying...`);
+            setTimeout(() => this.updateApiCredentials(), 200);
         }
+    }
+
+    async updateApiPermissions() {
+        return new Promise(async (resolve) => {
+            this.permissions = (await new MagisterApiRequestAccount().get())?.Groep?.[0]?.Privileges?.filter(p => p.AccessType.includes('Read')).map(p => p.Naam);
+            resolve(this.permissions);
+        });
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)